utils/land-patterns.stanza

```diff
@@ -5,281 +5,406 @@ defpackage ocdb/land-patterns :
+
+public defn round-rect-pth-pad (drill-r:Double, pad-w:Double, pad-h:Double, pad-r:Double) :
+  round-rect-pth-pad(C, drill-r, drill-r, pad-w, pad-h, pad-r)
+
+public defn round-rect-smd-pad (anchor:Anchor, w:Double, h:Double, r:Double) :
+  generic-pad(RoundedRectangle(anchor, w, h, r))
+
+public defn round-rect-smd-pad (w:Double, h:Double, r:Double) :
+  round-rect-smd-pad(C, w, h, r)
+
+public defn chamfered-rect-pth-pad (anchor:Anchor, drill-w:Double, drill-h:Double, pad-w:Double, pad-h:Double, pad-r:Double) :
+  val shape = chamfered(pad-w, pad-h, pad-r)
+  val hole  = capsule(anchor drill-w, drill-h)
+  generic-pad(shape, offset(shape,  SOLDER-MASK-REGISTRATION), hole, TH)
+
+public defn chamfered-rect-pth-pad (drill-r:Double, pad-w:Double, pad-h:Double, pad-r:Double) :
+  chamfered-rect-pth-pad(C, drill-r, drill-r, pad-w, pad-h, pad-r)
+
+public defn chamfered-rect-pth-pad (drill-w:Double, drill-h:Double, pad-w:Double, pad-h:Double, pad-r:Double) :
+  chamfered-rect-pth-pad(C, drill-w, drill-h, pad-w, pad-h, pad-r)
+
+public defn chamfered-rect-smd-pad (w:Double, h:Double, r:Double) :
+  generic-pad(chamfered(w, h, r))
+
+;==============================================================================
+;========================= Land Patterns ======================================
+;==============================================================================
+; Create a non-plated hole of arbitrary shape for hole and mask
+public pcb-landpattern non-plated-hole-landpattern (hole:Shape, mask:Shape):
+  layer(Cutout())           = hole
+  layer(SolderMask(Top))    = mask
+  layer(SolderMask(Bottom)) = mask
+  val d = dims(hole)
+  layer(Courtyard(Top))    = Rectangle(x(d), y(d))
+  layer(Courtyard(Bottom)) = Rectangle(x(d), y(d))
+
+; Create a non-plated hole landpattern with a hole shape
+public defn non-plated-hole-landpattern (hole:Shape):
+  non-plated-hole-landpattern(hole, offset(hole,  SOLDER-MASK-REGISTRATION))
+
+; Create a circular non-plated hole landpattern with given drill radius and mask radius
+public defn non-plated-hole-landpattern (drill-r:Double, mask-r:Double) :
+  non-plated-hole-landpattern(Circle(drill-r), Circle(mask-r))
+
+; Create a non-plated hole landpattern with a given drill radius.
+public defn non-plated-hole-landpattern (drill-r:Double) :
+  non-plated-hole-landpattern(drill-r, drill-r)
+
+;==============================================================================
+;========================== Land Pattern Generators ===========================
+;==============================================================================
+
+;==============================================================================
+;========================== Pose Arrangement Helpers ==========================
+;==============================================================================
+; PinPadMapping is a not-yet-placed pad, with assigned name, pad, and pose
+public pcb-struct ocdb/land-patterns/PinPadMapping:
```

Comment:
The reified type `PinPadMapping` and helper `make-generic-landpattern` is too sophisticated to expose publicly. It seems like they are for internal use, so please make them private.